test/unit/session/test_session.py

```diff
@@ -221,6 +222,42 @@ def test_box_session_raises_for_failed_response(box_session, mock_network_layer,
         box_session.get(url=test_url)
 
 
+def test_box_session_retries_requests_library_exceptions(box_session, mock_network_layer, generic_successful_request_response, test_url):
+    mock_network_layer.request.side_effect = [RequestException(), generic_successful_request_response]
+    mock_network_layer.retry_after.side_effect = lambda delay, request, *args, **kwargs: request(*args, **kwargs)
+
+    box_response = box_session.get(url=test_url)
+    assert box_response.status_code == 200
+
+
+def test_box_session_raises_requests_library_exception_when_max_retries_limit_is_reached(box_session, mock_network_layer, test_url):
+    mock_network_layer.request.side_effect = [RequestException()] * 6
```

Comment:
Don't you want to use API.MAX_RETRY_ATTEMPTS + 1 instead of hardcoded 6?

Reply:
makes sense